src/Endpoints/Zones.php

```diff
@@ -228,10 +228,10 @@ public function cachePurgeEverything(string $zoneID): bool
         return false;
     }
 
-    public function cachePurge(string $zoneID, array $files = null, array $tags = null, array $hosts = null): bool
```

Comment:
the cyclomatic complexity of this method hit 10 with these additions, causing CI to fail. Some functionality will need to be extracted into smaller methods.